pkg/controller/restore_controller.go

```diff
@@ -255,8 +259,31 @@ func (c *restoreController) processRestore(key string) error {
 	// don't modify items in the cache
 	restore = restore.DeepCopy()
 
-	// complete & validate restore
-	if restore.Status.ValidationErrors = c.completeAndValidate(restore); len(restore.Status.ValidationErrors) > 0 {
+	excludedResources := sets.NewString(restore.Spec.ExcludedResources...)
+	for _, nonrestorable := range nonRestorableResources {
+		if !excludedResources.Has(nonrestorable) {
+			restore.Spec.ExcludedResources = append(restore.Spec.ExcludedResources, nonrestorable)
+		}
+	}
+
+	var (
+		fetchErr error
+		backup   *api.Backup
+	)
+	// attempt to get a schedule name whether provided directly or through a backup
+	backupScheduleName := restore.Spec.ScheduleName
+	if backupScheduleName == "" {
+		backup, fetchErr = c.fetchBackup(c.bucket, restore.Spec.BackupName)
+		if backup != nil {
+			backupScheduleName = backup.GetLabels()["ark-schedule"]
+		}
+	}
+
+	// Register attempts before we do validation so we can get better tracking
```

Comment:
What does "better tracking" actually mean here?  IOW, what would be the problem if we move the call to `RegisterRestoreAttempt` to below `completeAndValidate`?

Reply:
Good question - essentially, I wanted to increment attempts before incrementing any other type of restore metric. I believe that's doable after calling into `completeAndValidate`, though, since it would just register an attempt then register a failed validation.